.github/workflows/deploy.yml

```diff
@@ -0,0 +1,19 @@
+name: deploy
+
+on:
+  push:
+    branches:
+      - 'master'
+
+jobs:
+  deploy:
+    runs-on: ubuntu-18.04
+    steps:
+    - uses: actions/checkout@v1
+    - name: Setup Java
+      uses: actions/setup-java@v1
+      with:
+        java-version: '1.8'
+        architecture: x64
+    - name: Deploy
+      run: ./deploy-snapshot.sh
```

Comment:
I want this workflow to be executed after all jobs in the `main` workflow have been finished successfully.
I think keeping workflows separated has advantages of maintainability and security but it seems we can't define dependencies for workflows for now.
Do we have to merge this job into `main` with [`if`](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idif) directive?

Reply:
Yes. As far as I know, there's no way except `if` directive.
I incorporated the `deploy` job into the `main` workflow with some constraints. When pushing master, if any of `main` tests fails, the deploy is not executed.